### PR TITLE
Build and publish OLM bundle

### LINKED ISSUE #125: Build OLM bundle on every merge
### What problem are you facing?

Currently, building the OLM bundle is a manual process. While builds should be reproducible from source, we are not currently building until it is actually time to publish a version.

### How could Universal Crossplane help solve your problem?

UXP CI pipelines should build and publish OLM bundles alongside other artifacts.

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Updates OLM bundle build and adds a publish step so that we publish the
bundle alongside other artifacts.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Fixes #125 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

`make build`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -138,13 +138,19 @@ generate-chart: crossplane
...
+olm.build: $(HELM) $(OLMBUNDLE)
 	@$(INFO) Generating OLM bundle
 	@$(HELM) -n upbound-system template $(HELM_CHARTS_DIR)/$(PACKAGE_NAME) --set upbound.controlPlane.permission=edit > $(WORK_DIR)/olm.yaml
 	@$(SED_CMD) 's|RELEASE-NAME|$(PROJECT_NAME)|g' $(WORK_DIR)/olm.yaml
 	@rm -rf $(OLM_DIR)/bundle
 	@cat $(WORK_DIR)/olm.yaml | $(OLMBUNDLE) --version $(HELM_CHART_VERSION) --chart-file-path $(HELM_CHARTS_DIR)/$(PACKAGE_NAME)/Chart.yaml --extra-resources-dir $(CRDS_DIR) --output-dir $(OLM_DIR)
 
+olm.artifacts: olm.build
+	@mkdir -p $(abspath $(OUTPUT_DIR)/olm/$(VERSION))
+	@cp -r $(OLM_DIR)/bundle/. $(abspath $(OUTPUT_DIR)/olm/$(VERSION))
```
Comment (hasheddan) [inline Makefile:150]: I think it might make more sense to put the contents here into a tarball to publish.

File: cluster/olm/README.md [hunk 2]
```diff
@@ -2,30 +2,42 @@
...
+OperatorHub, download the subdirectory named `$VERSION` in the `olm` directory
+and open two separate PRs to the following directories with it copied into the
+specified path:
+* https://github.com/operator-framework/community-operators/tree/master/community-operators/universal-crossplane
+* https://github.com/operator-framework/community-operators/tree/master/upstream-community-operators/universal-crossplane
+
+The bundle output is specified by the contents of the helm chart and can be
+modified by altering the following files. Any changes made to these files should
+be committed to the repository before tag and release of a new version.
 * `Chart.yaml` is used for metadata, so make sure it contains the new version.
```
Comment (muvaf) [inline cluster/olm/README.md:22]: nitpick: It sounds like all three files are in the same directory but they aren't. Maybe talk about this one separately?

File: Makefile (hunk 1 above)
Comment (muvaf) [reply Makefile:150]: Does build submodule take care of uploading whatever we put under `$(OUTPUT_DIR)`?

**Review by muvaf — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (hasheddan) [reply Makefile:150]: It does 👍 You can see examples of this in `up` where I put the debs and rpm in the output dir https://cli.upbound.io/p?prefix=stable/v0.2.0/

Comment (muvaf) [reply Makefile:150]: Got it. Tarball makes sense to me, too. Folder downloading may not be as straightforward.

**Pull request merged**